src/main/java/de/tum/in/ase/parser/strategy/GCCParser.java

```diff
@@ -0,0 +1,154 @@
+import de.tum.in.ase.parser.domain.Report;
+
+public class GCCParser implements ParserStrategy {
+
+    // Locations in regex group
+    protected static final int NO_SEGS = 3;
+
+    protected static final int FILE_POS = 1;
+
+    protected static final int ROW_POS = 2;
+
+    protected static final int COL_POS = 3;
+
+    protected static final int TYPE_POS = 4;
+
+    protected static final int DESC_POS = 5;
+
+    protected static final int ERROR_POS = 6;
+
+    protected static final int BODY_SEG = 2;
+
+    // Categories
+    protected static final String MEMORY = "Memory";
+
+    protected static final String BAD_PRACTICE = "BadPractice";
+
+    protected static final String SECURITY = "Security";
+
+    protected static final String UNDEFINED_BEHAVIOR = "UndefinedBehavior";
+
+    protected static final String MISC = "Misc"; // For various other results, that are not part of the static analysis
+
+    // Map that contains the matching category for each error
+    protected static final Map<String, String> categories = new HashMap<>();
+
+    public Report parse(Document doc) {
+        Report report = new Report(StaticCodeAnalysisTool.GCC);
+        extractIssues(doc, report);
+        return report;
+    }
+
+    /**
+     * Constructs issues and adds them to the static analysis report.
+     *
+     * @param doc Contains the output from GCC SCA
+     * @param report The report the issues will be added to
+     */
+
+    protected void extractIssues(Document doc, Report report) {
+        initCategoryMapping();
+        List<Issue> issues = new ArrayList<>();
+        Element gccLog = doc.getDocumentElement();
+
+        // Splits up entries by the following format: <filename>:<row>:<col>: <description>
+        // ?= is a look ahead regex, so we can keep the delimiter after the split
+        String delim = "(?=(\\n(.)+.\\w:(\\d)+:(\\d)+:(.)+:(.)+))";
+        String[] results = gccLog.getTextContent().split(delim);
+
+        // For simplicity, we use \n also as the delimiter, so we only get one actual match for each entry
+        String infoEntryRegex = "([^:^\\n]+):(\\d+):(\\d+):\\s(\\w+\\s*\\w*):\\s(.+)(\\[.+\\])\\n";
+        Pattern errorDetails = Pattern.compile(infoEntryRegex);
+        Matcher m = errorDetails.matcher("");
+
+        for (String entry : results) {
+            Issue issue = new Issue(null);
+
+            m.reset(entry);
+
+            // Extract attributes
+            if (m.find()) {
+                boolean isAnalyzerIssue;
+                String filename = m.group(FILE_POS).trim();
+                Integer row = Integer.parseInt(m.group(ROW_POS));
+                Integer col = Integer.parseInt(m.group(COL_POS));
+                String type = m.group(TYPE_POS);
+                String description = m.group(DESC_POS);
+                String errorName = m.group(ERROR_POS);
+                String body;
+
+                // Extract output after info line
+                if (errorName != null) {
```

Comment:
By inverting the if condition to an
```java
String errorName = m.group(ERROR_POS);
if (errorName == null) {
    continue;
}
```
the other declarations can be moved down and one less indentation level is needed.

Reply:
I applied this in the latest commit.